hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -2033,7 +2024,11 @@
       name: chip_sw_csrng_lc_hw_debug_en
       desc: '''Verify the effect of LC HW debug enable on CSRNG.
 
-            TODO: This is pending SCA security review and might be removed.
+            lc_hw_debug_en is used to diversify the csrng seed.
+            - Instantiate CSRNG with fixed seed and fetch entropy while in TEST state (hw_debug_en = 1).
+            - Save the retrieved entropy in flash and advance the device to PROD or DEV state.
+            - Instantiate CSRNG with the same fixed seed and fetch entropy again while in the NEW state.
+            - The newly fetched entropy and the old entropy should not match.
```

Comment:
Perhaps a better verification is to positively match to the NonProduction Constant ^ with seed1. Then test the Production Constant ^ seed2. I also think this requires a lot of RTL peeking, since registers cannot observe values directly. Maybe "fetch entropy" means a csrng gen command is run and the returned gen bits are compared against.
If the two ES seeds were always consistently generated, and the diversification constants were known, then the resulting gen bits could be pre-computed and then compared against. Point is from this description, can't tell how this test will work.

Reply:
that would do it, but seems like a lot of RTL to change to test one bit (hw_debug). Also, seems like forcing the ES seed to something specific (like zero, for example) might be a security risk. We did test the seed mux in csrng at unit text, so not sure how best to test the hw_debug mode at a chip level.

Reply:
o wait, sorry i wasn't thinking about changing the RTL. I was thinking about using the entropy_src injection flow (similar to the known answer test).  So from there, using software, I want to inject a "known" value into the conditioner, have that value then feed into csrng, and then use the lc_hw_debug_en to change the expectation of csrng.

Although... this does seem like overkill over all.. since most of the functionality is probably covered in block level.

Reply:
wow...i forgot about that function for ES...I wonder who designed that block...I actually like this suggestion, and yes overkill all over, but testing KA function in two units together. It might actually be valuable, beyond just testing hw_debug. If this was all spelled out in the test item (or a little bit more explanation), that might be the way to go.

Reply:
sounds good. I can do that.
@moidx / @sriyerg any objections?

Reply:
This sounds like a lot of work. Assuming the diversification is tested at the IP level, maybe this is something we can cover with a connectivity test?

If we want to implement this test, based on the discussion, then the procedure would be as follows:

1. Disable entropy_src, csrng and edn (in that order).
2. Enable entropy_src in FW_OVR mode. Maybe consider disabling the conditioner to help
    in pre-calculations.
3. Instantiate SW CSRNG with flag0=0. 
4. Run test similarly to a KAT test but injecting data from entropy_src fw override buffer.

This test could be useful, but not sure if the overhead of pre-calculating all the expected values based on diversification would be worth the effort. A simpler effort would be to check that the values don't match across lc_states?

Reply:
That all sounds correct.

Reply:
ah yes @moidx i should have clarified. My only goal was to check that they did not match.. (perhaps preceded by a case that when not in PROD state they DO match).

Reply:
@mwbranstad / @moidx i've updated the test description, let me know if this looks good to you.

Reply:
@mwbranstad can you also confirm one thing? `lc_hw_debug_en` **DOES NOT** impact the reading of the csrng internal state right? I think we specifically moved that to a different OTP bit just in case. I wanted to confirm with you though.

Reply:
also discussed with @sriyerg offline, and he agrees we should keep a functional test instead of resorting to connectivity test.

Reply:
@tjaychen confirmed - lc_hw_debug_en only selects the seed modification that is received from ES.